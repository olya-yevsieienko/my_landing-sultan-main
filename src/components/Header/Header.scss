@import '../../styles/extends.scss';
@import '../../styles/mixin.scss';
@import '../../styles/variables.scss';

.header {
  position: relative;

  &__content {
    &::before {
      content: '';
      position: absolute;
      top: $header;
      right: 0;
      left: 0;
      display: block;
      width: 100%;
      height: 1px;
      background: $main;
      opacity: 0.2;
    }

    &::after {
      content: '';
      bottom: 0;
      right: 0;
      left: 0;
      display: block;
      width: 100%;
      height: 1px;
      background: $main;
      opacity: 0.2;
    }
  }
  
  &__main {
    @extend %position-content;
    display: flex;
    justify-content: space-between;
    align-items: center;
    height: $header;

    @include onPortraitTablet {
      height: 108px;
    }
  }

  &__mobile-menu {
    position: relative;
    display: flex;
    justify-content: center;
    align-items: center;
    height: 32px;
    width: 32px;
    border-radius: 50%;
    background: $button;

    @include onPortraitTablet {
      display: none;
    }
  }

  &__logo {
    display: block;
    width: 98x;
    height: 41px;

    @include onPortraitTablet {
      height: 66px;
      width: 156px;
    }

    @include onTablet {
      margin-right: 20px;
    }

    @include onDesktop {
      margin-right: 38px;
    }
  }

  &__catalog {
    display: none;

    @include onTablet {
      margin-right: 6%;
      display: flex;
      height: 60px;
      width: 160px;
    }
    
    @include onDesktop {
      margin-right: 15px;
      display: flex;
      height: 60px;
      width: 14%;
    }

    &--btn-name {
      margin-right: 12px;
    }
  }

  &__search {
    display: none;

    @include onPortraitTablet {
      display: block;
      width: 200px;
    }

    @include onDesktop {
      margin-right: 2.5%;
      width: 19%;
    }
  }

  &__contact {
    display: none;

    @include onPortraitTablet {
      display: flex;
      flex-direction: column;
      align-items: end;
    }
  }

  &__call-order {
    display: none;

    @include onDesktop {
      position: relative;
      margin-right: 3.6%;
      display: flex;
      align-items: center;
      text-align: right;

      &::after {
        content: '';
        position: absolute;
        top: 26px;
        right: 6px;
        display: block;
        height: 8px;
        width: 8px;
        border: 2px solid #F0F6FA;
        border-radius: 50%;
        background: #1FD85D;
      }
    }
  }

  &__number {
    margin-bottom: 5px;
    font-family: 'InterSemiBold', serif;
  }

  &__schedule {
    margin-bottom: 5px;
    font-family: 'InterLight', serif;
    font-size: 12px;
  }

  &__call {
    width: 81px;
    font-size: 10px;
    border-bottom: 1px solid $main;
    cursor: pointer;
  }

  &__consultant {
    display: none;

    @include onDesktop {
      margin-left: 10px;
      display: block;
    }
  }

  &__price-list {
    display: none;

    @include onDesktop {
      margin-right: 3.8%;
      display: flex;
      height: 60px;
      width: 15%;
      min-width: 120px;
    }

    &--btn-name {
      margin-right: 14px;
    }
  }

  &__basket {
    display: flex;
    align-items: center;
  }

  &__img-basket {
    position: relative;
    display: block;
    height: 16px;
    width: 22px;
    background: url(../../image/icons/basket.svg);
    background-size: contain;
    background-repeat: no-repeat;

    @include onPortraitTablet {
      margin-right: 28px;
    }
  }

  &__basket-items {
    position: absolute;
    top: -10px;
    right: -7px;
    display: flex;
    justify-content: center;
    align-items: center;
    height: 18px;
    width: 18px;
    font-size: 10px;
    font-weight: bold;
    color: #fff;
    border: 3px solid #fff;
    border-radius: 50%;
    background-color: $button;
  }

  &__descrpt {
    display: none;

    @include onPortraitTablet {
      display: flex;
      flex-direction: column;
      width: 72px;
    }
  }

  &__basket-name {
    font-family: 'InterLight', serif;
    font-size: 12px;
    line-height: 150%;
  }

  &__amount {
    font-family: 'InterSemiBold', serif;
    line-height: 150%;
    color: $secondary;
  }

  &__mobile-head {
    position: relative;
    display: flex;
    justify-content: center;

    @include onPortraitTablet {
      display: none;
    }
  }

  &__mobile-btns {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 40px;
    width: 160px;
    cursor: pointer;
  }

  &__icon {
    margin-right: 6px;
    display: block;
    
    &--catalog {
      height: 15px;
      width: 15px;
      background: url('../../image/icons/catalog.svg');
    }

    &--search {
      height: 12px;
      width: 12px;
      background: url('../../image/icons/search.svg');
      background-repeat: no-repeat;
      background-size: contain;
    }
  }

  &__btn-name {
    font-family: 'InterSemiBold', serif;
    font-size: 12px;
  }

  &__mobile-line {
    position: absolute;
    top: 5px;
    height: 30px;
    border-right: 1px dashed $main;
    opacity: 0.1;
  }
}