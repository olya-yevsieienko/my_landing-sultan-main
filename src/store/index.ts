import { combineReducers, configureStore } from '@reduxjs/toolkit';
import cartReducer from './cartSlice';
import filterSlice from './filterSlice';
import goodsSlice from './goodsSlice';

const rootReducer = combineReducers({
  cart: cartReducer,
  goods: goodsSlice,
  filter: filterSlice,
});

const store = configureStore({
  reducer: rootReducer,
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
