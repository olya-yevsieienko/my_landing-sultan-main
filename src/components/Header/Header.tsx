import React, { useState } from 'react';
import { Link, useNavigate } from 'react-router-dom';
import { useAppSelector } from '../../hooks/redux';

import { Navigation } from './Navigation';
import { CustomButton } from '../UI/CustomButton';
import { CustomSearch } from '../UI/CustomSearch/CustomSearch';
import { MobileMenu } from './MobileMenu';

import logo from '../../image/icons/logo.svg';
import catalog from '../../image/icons/catalog-white.svg';
import price from '../../image/icons/price-list.svg';
import consultant from '../../image/consultant.png';
import menuClosed from '../../image/icons/mobile-menu.svg';
import menuOpened from '../../image/icons/menu-opened.svg';
import './Header.scss';
import { getRoundedNum } from '../../utils/getRoundedNum';

const Header: React.FC = () => {
  const [isMenuOpen, setIsMenuOpen] = useState(false);
  const { totalAmount, totalCount } = useAppSelector((state) => state.cart);
  const navigate = useNavigate();

  const handleRouteChange = (path: string) => {
    const newPath = path;
    navigate(newPath);
  };

  return (
    <header className="header">
      <Navigation />

      <div className="header__content">
        <div className="header__main">
          <div
            className="header__mobile-menu"
            onClick={() => setIsMenuOpen(!isMenuOpen)}
          >
            <img src={isMenuOpen ? menuOpened : menuClosed} alt="" />
          </div>

          {isMenuOpen && <MobileMenu />}

          <Link to={'/catalog'}>
            <img src={logo} alt="Султан" className="header__logo" />
          </Link>

          <div className="header__catalog">
            <CustomButton
              title={'Каталог'}
              imageSrc={catalog}
              addedClass={'header__catalog--btn-name'}
              onClick={() => handleRouteChange('/catalog')}
            />
          </div>

          <div className="header__search">
            <CustomSearch />
          </div>

          <div className="header__call-order">
            <div className="header__contact">
              <span className="header__number">+7 (777) 490-00-91</span>
              <span className="header__schedule">время работы: 9:00-20:00</span>
              <span className="header__call">Заказать звонок</span>
            </div>
            <img className="header__consultant" src={consultant} alt="" />
          </div>

          <div className="header__price-list">
            <CustomButton
              title={'Прайс-лист'}
              imageSrc={price}
              addedClass={'header__price-list--btn-name'}
            />
          </div>

          <div className="header__basket">
            <Link to="/basket">
              <div className="header__img-basket">
                <span className="header__basket-items">{totalCount}</span>
              </div>
            </Link>
            <div className="header__descrpt">
              <span className="header__basket-name">Корзина</span>
              <span className="header__amount">
                {getRoundedNum(totalAmount)} ₸
              </span>
            </div>
          </div>
        </div>

        <div className="header__mobile-head">
          <Link to="/catalog">
            <div className="header__mobile-btns">
              <div className="header__icon header__icon--catalog"></div>
              <span className="header__btn-name">Каталог</span>
            </div>
          </Link>

          <div className="header__mobile-line"></div>

          <div className="header__mobile-btns">
            <div className="header__icon header__icon--search"></div>
            <span className="header__btn-name">Поиск</span>
          </div>
        </div>
      </div>
    </header>
  );
};

export default Header;
