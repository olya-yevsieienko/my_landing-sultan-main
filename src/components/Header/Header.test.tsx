import { screen, render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../store';
import Header from './Header';

describe('Header testing', () => {
  test('Header rendering was successful', () => {
    expect(<Header />).toMatchSnapshot();
  });
  test('Header displays the correct total amount of the empty cart', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      </Provider>
    );
    expect(screen.getByTestId('header-amount')).toHaveTextContent('0');
  });
});
