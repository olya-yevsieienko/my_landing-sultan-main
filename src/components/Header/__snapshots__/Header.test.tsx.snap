// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header testing Header rendering was successful 1`] = `<Header />`;
